lib/pdf/core/renderer.rb

```diff
@@ -216,10 +217,12 @@ def render_xref(output)
       # Write out the PDF Trailer, as per spec 3.4.4
       #
       def render_trailer(output)
+        trailer_id = PDF::Core::ByteString.new(SecureRandom.random_bytes(16))
```

Comment:
Indeed, the time is mentioned. But it doesn't seem like a good indicator of content stability. After all, are two documents different if their content is identical? Also a bit earlier in that chapter one can find this:

> The first byte string is a permanent identifier based on the contents of the file at the time it was originally created and does not change when the file is incrementally updated. The second byte string is a changing identifier based on the file’s contents at the time it was last updated. When a file is first written, both identifiers are set to the same value.

I would just hash the body. It would work great for our purpose. Also since Prawn doesn't really support incremental updates we can reuse the value for the second part of ID.